common/lib/xmodule/xmodule/course_module.py

```diff
@@ -910,6 +918,48 @@ def __init__(self, *args, **kwargs):
             self.discussion_topics = {_('General'): {'id': self.location.html_id()}}
 
         try:
+            enrollment_track_scheme = UserPartition.get_scheme("enrollment_track")
+            enrollment_track_partitions = self.get_user_partitions_for_scheme(enrollment_track_scheme)
+            # Note that when a course_module is initially created, the ID is the draft version.
+            # We just want to store the course_id without published or draft.
+            course_id = unicode(self.id.for_branch(None))
+            create_partition = False
+            new_id = 0
+
+            if len(enrollment_track_partitions) == 0:
+                used_ids = set(p.id for p in self.user_partitions)
+                while new_id in used_ids:
+                    new_id += 1
+                create_partition = True
```

Comment:
Yes, if this is called by LMS it will update the course module. But the init method in general sets up a lot of default values, so doesn't the same situation occur for those? Let's chat.